src/python/pants/option/global_options.py

```diff
@@ -903,7 +903,7 @@ class BootstrapOptions:
     )
     pants_ignore = StrListOption(
         advanced=True,
-        default=[".*/", _default_rel_distdir, "__pycache__"],
+        default=[".*/", _default_rel_distdir, "__pycache__", "!.semgrep/"],
```

Comment:
As discussed, this is applied globally, not just when the semgrep backend is enabled.